src/compiler/transformers/decorators-to-static/convert-decorators.ts

```diff
@@ -414,8 +414,20 @@ export const updateConstructor = (
  * @param classDeclaration a class declaration AST node
  * @returns whether this class has parents or not
  */
-const needsSuper = (classDeclaration: ts.ClassDeclaration): boolean =>
-  classDeclaration.heritageClauses && classDeclaration.heritageClauses.length > 0;
+const needsSuper = (classDeclaration: ts.ClassDeclaration): boolean => {
+  let hasHeritageClauses = classDeclaration.heritageClauses && classDeclaration.heritageClauses.length > 0;
+
+  if (hasHeritageClauses) {
+    // A {@link ts.SyntaxKind.HeritageClause} node may be for extend a
+    // superclass _or_ for implementing an interface. We only want to add a
+    // `super()` call to our synthetic constructor here in the case that there
+    // is a superclass, so we can check for that situation by checking for the
+    // present of a heritage clause with the `.token` property set to
+    // `ts.SyntaxKind.ExtendsKeyword`.
```

Comment:
TIL

Reply:
me too 🤦‍♀️

Reply:
Kinda dumb imo. They should offer a better way of checking this 🙄